src/hooks/usePendingSafeStatuses.ts

```diff
@@ -0,0 +1,134 @@
+import { SafeCreationEvent, safeCreationSubscribe } from '@/features/counterfactual/services/safeCreationEvents'
+import {
+  PendingSafeStatus,
+  removeUndeployedSafe,
+  selectUndeployedSafe,
+  updateUndeployedSafeStatus,
+} from '@/features/counterfactual/store/undeployedSafesSlice'
+import { checkSafeActionViaRelay, checkSafeActivation } from '@/features/counterfactual/utils'
+import { isSmartContract, useWeb3ReadOnly } from '@/hooks/wallets/web3'
+import { useAppDispatch, useAppSelector } from '@/store'
+import { useEffect, useRef } from 'react'
+import useChainId from './useChainId'
+import useSafeInfo from './useSafeInfo'
+
+const pendingStatuses: Partial<Record<SafeCreationEvent, PendingSafeStatus | null>> = {
+  [SafeCreationEvent.PROCESSING]: PendingSafeStatus.PROCESSING,
+  [SafeCreationEvent.RELAYING]: PendingSafeStatus.RELAYING,
+  [SafeCreationEvent.SUCCESS]: null,
+  [SafeCreationEvent.FAILED]: null,
+  [SafeCreationEvent.REVERTED]: null,
+}
+
+const usePendingSafeMonitor = (): void => {
```

Comment:
Should this file be in features/counterfactual?

Reply:
Yes, eventually it makes sense to move it out when it will be reused for the safe creation flow but I would keep it in the features folder until then.